server/Dockerfile

```diff
@@ -17,7 +17,10 @@ ARG KEYCLOAK_DIST=https://downloads.jboss.org/keycloak/$KEYCLOAK_VERSION/keycloa
 
 USER root
 
-RUN yum update -y && yum install -y epel-release git && yum install -y jq openssl which && yum clean all
+RUN microdnf update -y && microdnf install -y dnf git hostname java-11-openjdk.x86_64 openssl tar which && microdnf clean all
```

Comment:
@douglaspalmer could we use the same just with `microdnf`, instead of use `dnf`. The reason why I'm asking, it's because dnf adds more Python dependencies. See:

dnf
package: dnf-4.2.2-2.fc30.noarch
  dependency: /bin/sh
   provider: bash-5.0.7-1.fc30.i686
   provider: bash-5.0.7-1.fc30.x86_64
  dependency: python3-dnf = 4.2.2-2.fc30
   provider: python3-dnf-4.2.2-2.fc30.noarch
  dependency: systemd
   provider: systemd-241-12.git1e19bcd.fc30.i686
   provider: systemd-241-12.git1e19bcd.fc30.x86_64

package: dnf-4.2.8-2.fc30.noarch
  dependency: /bin/sh
   provider: bash-5.0.7-1.fc30.i686
   provider: bash-5.0.7-1.fc30.x86_64
  dependency: python3-dnf = 4.2.8-2.fc30
   provider: python3-dnf-4.2.8-2.fc30.noarch
  dependency: systemd
   provider: systemd-241-12.git1e19bcd.fc30.i686
   provider: systemd-241-12.git1e19bcd.fc30.x86_64

Reply:
Git also adds Python dependencies, but there's nothing we can do about this. If there's a possibility of install these RPMs with just `microdnf`, that would be interesting.

@stianst does it make sense to you?

Reply:
Git is only used when image is built from Keycloak master so we can move installing git only when that's the case

Reply:
The whole idea of building image and allowing it to also build Keycloak from git was perhaps a bit hacky ;)

Reply:
@stianst so let me try to summarize what we want to achieve here:
- Get rid of `dnf` as a dependency to prevent adding Python to the Docker image
- Add checks to install `git` only when image is built from Keycloak master

Does it make sense?

Reply:
I have added another commit that removes dnf. In order to achieve this I had to add Fedora repositories to the repository config file. Is this acceptable? I can squash the commits if everyone is happy with this.

Reply:
While we are discussing dependencies, do we need to install jq? I can't find anywhere that we use it.